### PR TITLE
Use bulk conversion in Write8BitColor

### PR DESCRIPTION
**Pull request by brianpopow**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/SixLabors/ImageSharp/pulls) open
- [x] I have verified that I am following matches the existing coding patterns and practice as demonstrated in the repository. These follow strict Stylecop rules :cop:.
- [x] I have provided test coverage for my change (where applicable)

### Description

Just a small improvement in writing 8 bit bitmaps with a palette: Converting the quantized colors to rgb is now done as bulk convert.

### CODE REVIEW

File: src/ImageSharp/Formats/Bmp/BmpEncoderCore.cs [hunk 1]
```diff
@@ -340,15 +339,15 @@ private void Write8BitColor<TPixel>(Stream stream, ImageFrame<TPixel> image, Spa
         {
             using IQuantizer<TPixel> frameQuantizer = this.quantizer.CreatePixelSpecificQuantizer<TPixel>(this.configuration);
             using IndexedImageFrame<TPixel> quantized = frameQuantizer.BuildPaletteAndQuantizeFrame(image, image.Bounds());
+            using IMemoryOwner<Rgba32> rgbColorsBuffer = this.memoryAllocator.Allocate<Rgba32>(quantized.Palette.Length);
+            Span<Rgba32> rgbColors = rgbColorsBuffer.GetSpan();
 
             ReadOnlySpan<TPixel> quantizedColors = quantized.Palette.Span;
-            var color = default(Rgba32);
+            PixelOperations<TPixel>.Instance.ToRgba32(Configuration.Default, quantizedColors, rgbColors);
```
Comment (JimBobSquarePants) [inline src/ImageSharp/Formats/Bmp/BmpEncoderCore.cs:346]: This should be the configuration that the memory allocator is derived from.

File: src/ImageSharp/Formats/Bmp/BmpEncoderCore.cs [hunk 2]
```diff
@@ -340,15 +339,15 @@ private void Write8BitColor<TPixel>(Stream stream, ImageFrame<TPixel> image, Spa
...
 
-            // TODO: Use bulk conversion here for better perf
             int idx = 0;
-            foreach (TPixel quantizedColor in quantizedColors)
+            foreach (Rgba32 color in rgbColors)
```
Comment (antonfirsov) [inline src/ImageSharp/Formats/Bmp/BmpEncoderCore.cs]: Why not something like:
```C#
PixelOperations<TPixel>.Instance.ToRgba32(Configuration.Default, quantizedColors, MemoryMarshal.Cast<byte, Rgba32>(colorPalette));
ref Span<uint> colorPaletteAsUInt= MemoryMarshal.Cast<byte, uint>(colorPalette);
for (int i =0;i<colorPaletteAsUInt;i++)
  colorPaletteAsUInt[i] = colorPaletteAsUInt[i] & 0xFFFFFF00;
```

Comment (JimBobSquarePants) [reply src/ImageSharp/Formats/Bmp/BmpEncoderCore.cs]: Tidy!

Comment (brianpopow) [reply src/ImageSharp/Formats/Bmp/BmpEncoderCore.cs]: yeah that will do, it just needs to be ToBgra instead

Comment (JimBobSquarePants) [reply src/ImageSharp/Formats/Bmp/BmpEncoderCore.cs]: Which should be fast enough with the new shuffle intrinsics

**Review by JimBobSquarePants — APPROVED** [review]

**Pull request merged**